dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/rest/rest/client/user/MessageAndRestClientTestServlet.java

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+ * Copyright (c) 2019 ,2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.telemetry.internal_fat.apps.rest.rest.client.user;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+import io.openliberty.microprofile.telemetry.internal_fat.common.SpanDataMatcher;
+import io.openliberty.microprofile.telemetry.internal_fat.common.spanexporter.InMemorySpanExporter;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import javax.inject.Inject;
+import javax.servlet.annotation.WebServlet;
+import junit.framework.Assert;
+
+@WebServlet("/testMessageAndRestClientTestServlet")
+public class MessageAndRestClientTestServlet extends FATServlet {
+
+    private static final long MAX_WAIT = 40000; //millis
+    private static final long WAIT_INTERVAL = 50; //millis
+
+    private static Map<Span, Integer> spanAndExpectedNumber = new HashMap<Span, Integer>();
+
+    @Inject
+    private SimpleReactiveMessagingBean reactiveBean;
+
+    @Inject
+    InMemorySpanExporter inMemorySpanExporter;
+
+    @Test
+    public void testReactiveMessagingThatCallsJaxClient() throws InterruptedException {
+        List<String> results = getResultsWithWaitIfNeeded();
+
+        assertEquals(3, results.size());
+        assertEquals(3, results.size());
+        assertFalse(results.contains("LENGTH 8"));
+        assertFalse(results.contains("LENGTH 9!"));
+        assertTrue(results.contains("LENGTH 10!"));
+        assertTrue(results.contains("LENGTH 11!!"));
+        assertTrue(results.contains("LENGTH 12!!!"));
+    }
```

Comment:
This part is just testing what that the reactive messaging app worked as expected, right?

I would argue that we don't even want to run the tests for the telemetry spans if these checks don't pass, so we should combine the two methods and do these checks first.

Also, these checks could be replaced with just `assertThat(results, contains("LENGTH10!", "LENGTH 11!!", "LENGTH 12!!!"));` which will check the length and the contents and report the actual value of `results` if it's not as expected.